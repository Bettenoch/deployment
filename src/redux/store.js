import { configureStore } from '@reduxjs/toolkit';

import townSlice from './AppSlice';

const store = configureStore({
  reducer: {
    weather: townSlice,
  },
});

export default store;
