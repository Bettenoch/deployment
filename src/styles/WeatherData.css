.main-container {
  display: flex;
  justify-content: space-between;
  background-color: #6f90af;
  padding: 1rem 2rem;
}

.icons {
  display: flex;
  gap: 1rem;
}

.home-title {
  color: azure;
  font-size: 1.8rem;
}

.search-bar {
  width: max-content;
  height: max-content;
  padding: 1rem;
}

.world-container {
  background-color: #3d5a80;
}

.world-text {
  color: azure;
  font-size: 1.5rem;
  text-transform: uppercase;
}

.world-details {
  display: flex;
  justify-content: space-between;
  padding: 2rem;
}

.world-map {
  width: 20rem;
}

.country-info {
  display: flex;
  flex-wrap: wrap;
}

.country-weather {
  border: 1px solid rgb(233, 211, 211);
  display: flex;
  flex: 1;
  background-color: #98c1d9;
}

.country-weather:nth-child(even) {
  background-color: #648c94;
}

.hall {
  padding: 3em 4em;
  text-decoration: none;
  list-style: none;
}

.card {
  display: flex;
  gap: 2rem;
}

.card-bottom {
  display: flex;
  flex-direction: column;
  text-decoration: none !important;
  list-style: none;
}

.right-icon {
  width: 3em;
}

.country-name {
  align-items: flex-end;
  text-decoration: none !important;
  list-style: none !important;
  font-weight: 800;
  color: azure;
  text-transform: uppercase;
}

.temp-info {
  text-decoration: none;
  list-style: none;
  font-weight: 400;
  color: azure;
}

.temp-dt {
  text-decoration: none;
  list-style: none;
}

@media only screen and (max-width: 700px) {
  .main-container {
    flex-direction: column;
  }

  .country-weather {
    border: none;
    background-color: #98c1d9;
  }

  .country-weather:nth-child(even) {
    background-color: #648c94;
  }

  .world-map {
    max-width: 10rem;
  }

  .world-text {
    font-size: 1rem;
  }

  .img-map {
    max-width: 80px;
    max-height: 80px;
  }

  .hall {
    padding: 1em 1em;
  }

  .country-info {
    display: grid;
    grid-template-areas:
      "one  two"
      "three four"
      "five six"
      "seven seven";
  }

  .country-weather:nth-of-type(1) {
    grid-area: one;
  }

  .country-weather:nth-of-type(2) {
    grid-area: two;
  }

  .country-weather:nth-of-type(3) {
    grid-area: four;
  }

  .country-weather:nth-of-type(4) {
    grid-area: three;
  }

  .country-weather:nth-of-type(5) {
    grid-area: five;
  }

  .country-weather:nth-of-type(6) {
    grid-area: six;
  }

  .country-weather:nth-of-type(7) {
    grid-area: seven;
  }

  .card {
    flex-direction: column;
  }
}
