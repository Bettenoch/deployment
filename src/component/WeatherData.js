import { Link } from 'react-router-dom';
import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchData } from '../redux/AppSlice';
import '../styles/WeatherData.css';
import world from './assets/world.png';
import tanzania from './assets/tanzaniamap.gif';
import kenya from './assets/kenyamap.gif';
import ethiopia from './assets/ethiopiamap.gif';
import morocco from './assets/marococco map.gif';
import cameroon from './assets/camerounmap.gif';
import somalia from './assets/somaliamap.gif';
import rwanda from './assets/rwanda07map.gif';

const WeatherData = () => {
  const [search, setSearch] = useState('');
  const dispatch = useDispatch();
  const { weather, status, error } = useSelector((state) => state.weather);

  const countryMap = {
    Kenya: kenya,
    Tanzania: tanzania,
    Ethiopia: ethiopia,
    Cameroon: cameroon,
    Morocco: morocco,
    Somalia: somalia,
    Rwanda: rwanda,
  };

  const handleCountry = (e) => {
    setSearch(e.target.value);
  };

  const searchResult = weather.filter((data) => {
    const country = data.name.toLowerCase();
    const locate = search.toLowerCase();
    return country.includes(locate);
  });

  useEffect(() => {
    if (status === 'waiting') {
      dispatch(fetchData());
    }
  }, [dispatch]);

  if (status === 'waiting') {
    return <h1>Loading...</h1>;
  }

  if (error) {
    return <h1>Error fetching data</h1>;
  }

  return (
    <>
      <header className="main-container">
        <p className="home-title">WeatherChart</p>
        <input
          className="search-bar"
          type="text"
          placeholder="Search Country e.g kenya"
          value={search}
          onChange={handleCountry}
        />

      </header>
      <section className="world-container">
        <article className="world-details">
          <img src={world} className="world-map" alt="world" />
          <h2 className="world-text">World Weather</h2>
        </article>
      </section>
      <section className="country-info">
        {
      searchResult.map((country) => (
        <article key={country.name} className="country-weather">
          <Link to={`/city/${country.name}`} className="hall">
            <div className="card">
              <div className="weather-chart">
                {countryMap[country.name] && (
                  <img
                    src={countryMap[country.name]}
                    alt={country.name}
                    width={120}
                    height={120}
                  />
                )}

              </div>
              <div className="card-bottom">
                <img className="right-icon" src="https://img.icons8.com/ios/50/ffffff/circled-right.png" alt="Cloud" />
                <h1 className="country-name">{country.name}</h1>
                <p className="temp-info">
                  <span className="temp-dt">Temperature </span>
                  {Math.round(country.main.temp)}
                  {' '}
                  Â°C
                </p>

              </div>
            </div>
          </Link>
        </article>
      ))
    }
      </section>
    </>
  );
};

export default WeatherData;
