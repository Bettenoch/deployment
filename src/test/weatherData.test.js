import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import WeatherData from '../component/WeatherData';

describe('testing page elements', () => {
  const testData = {
    weather: [
      { main: { temp: 18.46 } },
      { name: 'Kenya' },
    ],
    Status: false,
    error: null,
  };

  it('Test weather data', () => {
    const weatherDetails = testData.weather;
    expect(weatherDetails).toEqual(testData.weather);
  });

  it('Should return weather status', () => {
    const weatherHd = testData.Status;
    expect(weatherHd).toBe(testData.Status);
  });
  it('should give correct country name ', async () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <WeatherData />
        </BrowserRouter>
      </Provider>,
    );
    const info = screen.findByText('World Weather');
    expect(info).toMatchSnapshot();
  });
});
