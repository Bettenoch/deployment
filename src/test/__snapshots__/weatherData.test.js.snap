// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing page elements should give correct country name  1`] = `Promise {}`;
